react-forms/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,35 @@
+import { FormEvent, useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    console.log({ username, password });
+  }
+
+  return (
+    <form action="" onSubmit={handleSubmit}>
+      <label htmlFor="">
+        Username:
+        <input
+          type="text"
+          name="username"
+          value={username}
+          onChange={(e) => setUsername(e.target.value)}
+        />
+      </label>
+      <label htmlFor="">
```

Comment:
If you wrap the input with `<label>` then you don't need to provide the `htmlFor` attribute. Once again, it saves typing!